### ISSUE TITLE
Add support for asciidoctorj for qute-web-asciidoc extension

### ISSUE THREAD

**Issue by mcruzdev** [quarkiverse/quarkus-qute-web#119]
We are currently developing an extension called `qute-web-asciidoc`, which utilizes a pure Java library behind the scenes to convert AsciiDoc to HTML. While the library is efficient and lightweight, it lacks some of the advanced features offered by AsciidoctorJ.

Our goal is to enable switching between both implementations, providing flexibility depending on the specific needs or feature requirements of the project.

**Comment by mkouba** [comment]
I'm not sure to be honest. We could create a separate extension - `qute-web-asciidoctorj` that would depend on `qute-web-asciidoc` and "deactivate" the default implementation based on `asciidoc-java`.

**Comment by gsmet** [comment]
I recently stumbled upon https://github.com/fluorumlabs/asciidocj which might be a good component for pure Java Asciidoc transformation.

What I actually liked (and what I was looking for) is that it has an actual parser and it seems to compare its output with the reference impl in the tests.

Now it doesn't look actually maintained but it might be a good lightweight alternative to using JRuby.

To give some context about what I was looking for: I want to implement a lightweight Asciidoc -> Markdown transformation and I was looking for components that might help.
The parser here could be a good base for this work except it actually already transforms things to HTML while parsing so I somehow need to change it to not be HTML-based.

/cc @jtama

**Comment by jtama** [comment]
Maybe this is what you're looking for : https://github.com/opendevise/downdoc

**Comment by jtama** [comment]
It's not java, but it does converts asciidoc to markdown.

**Comment by gsmet** [comment]
Yeah I know about this one but as you say, it's not Java :). And the code is far from being easily adaptable to Java.
